### PR TITLE
File upload & marked entities 

### PR DESCRIPTION
**Pull request by jamestjw**
- Implement file upload feature
- Allow the downloading of files
- Implement general marked entities page

### CODE REVIEW

File: public/login.php [hunk 1]
```diff
@@ -24,6 +24,7 @@
         $_SESSION["error_message"] = "Invalid email or Password!";
     } else {
         $_SESSION["current_user"] = $user;
+        $_SESSION["current_user_id"] = $user->id;
```
Comment (chris7neves) [inline public/login.php:27]: can't this be grabbed from the current_user? like we did for first_name?

File: public/index.php [hunk 2]
```diff
@@ -10,6 +10,10 @@
   <li>
     <a href="discussions.php"><strong>View</strong></a> - list all discussions
   </li>
+  <!-- Remove this -->
+  <li>
+    <a href="marked_entity_files.php?marked_entity_id=1"><strong>View</strong></a> - list all marked entity files
```
Comment (chris7neves) [inline public/index.php:15]: Should this be removed? (according to your remove this comment above it :P)

File: data/init.sql [hunk 3]
```diff
@@ -36,3 +36,25 @@ CREATE TABLE IF NOT EXISTS discussion_messages (
 	created_at TIMESTAMP,
 	updated_at TIMESTAMP
 );
+
+CREATE TABLE IF NOT EXISTS attachments (
+	id INT(11) UNSIGNED AUTO_INCREMENT PRIMARY KEY,
+	file_id VARCHAR(60),
+	file_content_type VARCHAR(15),
```
Comment (chris7neves) [inline data/init.sql:43]: Wouldnt this not follow 3NF rules since it can be derived from file_filename?

File: public/marked_entity_files.php [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+if (isset($_POST['submit'])) {
+    $marked_entity_file = new MarkedEntityFile();
+    $marked_entity_file->title = $_POST['title'];
+    $marked_entity_file->description = $_POST['description'];
+    $marked_entity_file->entity_id = $_POST['marked_entity_id'];
+    $marked_entity_file->user_id = $_POST['user_id'];
+
+    // Uploads folder needs to be created in the public/ directory
+    // TODO: Make this more convenient
+    $target_dir = "uploads/";
```
Comment (chris7neves) [inline public/marked_entity_files.php:21]: should it be in a class/section subdir in uploads?

File: public/index.php (hunk 2 above)
Comment (jamestjw) [reply public/index.php:23]: 😂 I think I was typing something and then I got interrupted and forgot about this,  I meant to say something along the lines of "remove this when marked entity page is ready"

File: public/login.php (hunk 1 above)
Comment (jamestjw) [reply public/login.php:47]: yeah but I got this very annoying problem about how the User class hasn't been loaded yet and the entire thing didnt work at all, you can try fixing this problem

Comment (jamestjw) [reply public/login.php:47]: I did this for now to get things moving

File: public/marked_entity_files.php (hunk 4 above)
Comment (jamestjw) [reply public/marked_entity_files.php:21]: assuming that file_id is some unique ID corresponding to each file, I thought that it'd be okay to do this

File: public/marked_entities.php [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+        <div class="form-group">
+            <!-- TODO: Figure out how to support multiple file uploads -->
+            <label for="marked_entity_file">File</label>
+            <input type="file" class="form-control-file" id="marked_entity_file" name="marked_entity_file">
+        </div>
+
+        <div class="form-group">
+            <label for="due_at">Due at</label>
+            <input type="text" id="due_at" name="due_at">
```
Comment (drouba) [inline public/marked_entities.php:86]: Could this input type be a  date rather than just a text? To make sure its valid and not just random words?

**Review by drouba — APPROVED** [review]
All good from what I see 👍

**Pull request merged**